### PR TITLE
Feat/integrate wmlock on file operations

### PR DESCRIPTION
**Pull request by cnlangzi**
### Changes
- integrated `WriteLock` on `delete` command
- integrated `WriteLock` on `copy` command
- integrated `WriteLock` on `rename` command
- integrated `WriteLock` on `move` command 

### Fixes
-

### Tests
Tasks to complete before merging PR:
- [ ]  Ensure system tests are passing. If not [Run them manually](https://github.com/0chain/gosdk/actions/workflows/system_tests.yml) to check for any regressions :clipboard:
- [ ]  Do any new system tests need added to test this change? do any existing system tests need updated? If so create a PR at [0chain/system_test](https://github.com/0chain/system_test)
- [ ]  Merge your system tests PR to master AFTER merging this PR

### Associated PRs (Link as appropriate):
- gosdk: https://github.com/0chain/gosdk/pull/408
- zbox: https://github.com/0chain/zboxcli/pull/186

### CODE REVIEW

**Comment by service-0chain** [comment]
Manual system tests [success] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | master  |
| gosdk  | feat/integrate_wmlock_on_file_operations  |
| zboxcli  | feat/integrate_wmlock_on_file_operations  |
| zwalletcli  | staging  |
| 0chain  | staging  |
| blobber  | staging  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | staging  |
| 0proxy  | staging  |
| 0search  | staging  |

File: zboxcore/sdk/copyworker.go [hunk 1]
```diff
@@ -108,6 +111,16 @@ func (req *CopyRequest) ProcessCopy() error {
 		return errors.New("Copy failed: Copy request failed. Operation failed.")
 	}
 
+	writeMarkerMutex, err := CreateWriteMarkerMutex(client.GetClient(), req.allocationObj)
+	if err != nil {
+		return fmt.Errorf("Delete failed: %s", err.Error())
```
Comment (peterlimg) [inline zboxcore/sdk/copyworker.go]: Delete failed -> Copy failed

File: zboxcore/sdk/copyworker.go [hunk 2]
```diff
@@ -108,6 +111,16 @@ func (req *CopyRequest) ProcessCopy() error {
...
+	}
+	err = writeMarkerMutex.Lock(context.TODO(), req.connectionID)
+	defer writeMarkerMutex.Unlock(context.TODO(), req.connectionID) //nolint: errcheck
+	if err != nil {
+		return fmt.Errorf("Delete failed: %s", err.Error())
```
Comment (peterlimg) [inline zboxcore/sdk/copyworker.go]: Delete failed -> Copy failed

File: zboxcore/sdk/renameworker.go [hunk 3]
```diff
@@ -105,6 +108,16 @@ func (req *RenameRequest) ProcessRename() error {
 		return errors.New("Rename failed: Rename request failed. Operation failed.")
 	}
 
+	writeMarkerMutex, err := CreateWriteMarkerMutex(client.GetClient(), req.allocationObj)
+	if err != nil {
+		return fmt.Errorf("Delete failed: %s", err.Error())
```
Comment (peterlimg) [inline zboxcore/sdk/renameworker.go]: Delete failed -> Rename failed

File: zboxcore/sdk/renameworker.go [hunk 4]
```diff
@@ -105,6 +108,16 @@ func (req *RenameRequest) ProcessRename() error {
...
+	}
+	err = writeMarkerMutex.Lock(context.TODO(), req.connectionID)
+	defer writeMarkerMutex.Unlock(context.TODO(), req.connectionID) //nolint: errcheck
+	if err != nil {
+		return fmt.Errorf("Delete failed: %s", err.Error())
```
Comment (peterlimg) [inline zboxcore/sdk/renameworker.go]: Delete failed -> Rename failed

**Review by peterlimg — CHANGES_REQUESTED** [review]

File: zboxcore/sdk/renameworker.go (hunk 4 above)
Comment (cnlangzi) [reply zboxcore/sdk/renameworker.go]: fixed

File: zboxcore/sdk/renameworker.go (hunk 3 above)
Comment (cnlangzi) [reply zboxcore/sdk/renameworker.go]: fixed

File: zboxcore/sdk/copyworker.go (hunk 2 above)
Comment (cnlangzi) [reply zboxcore/sdk/copyworker.go]: fixed

File: zboxcore/sdk/copyworker.go (hunk 1 above)
Comment (cnlangzi) [reply zboxcore/sdk/copyworker.go]: fixed

**Review by peterlimg — APPROVED** [review]
lgtm

**Pull request merged**